src/objectExplorer/objectExplorerProvider.ts

```diff
@@ -29,29 +31,36 @@ export class ObjectExplorerProvider implements vscode.TreeDataProvider<any> {
         return node;
     }
 
-    getChildren(element?: TreeNodeInfo): Promise<vscode.TreeItem[]> {
-        const children = this._objectExplorerService.getChildren(element);
+    async getChildren(element?: TreeNodeInfo): Promise<vscode.TreeItem[]> {
+        const children = await this._objectExplorerService.getChildren(element);
         if (children) {
-            return Promise.resolve(children);
+            return children;
         }
     }
 
-    async createSession(): Promise<string> {
-        return await this._objectExplorerService.createSession();
+    async createSession(promise: Deferred<TreeNodeInfo>, connectionCredentials?: IConnectionCredentials): Promise<void> {
+        return this._objectExplorerService.createSession(promise, connectionCredentials);
     }
 
     public getConnectionCredentials(sessionId: string): ConnectionCredentials {
```

Comment:
>sessionId [](start = 36, length = 9)

This should be sessionId? if it's actually expected to be possibly undefined

Reply:
And why are you even doing the check here? The OE Service already does this check so this shouldn't be necessary. 

---
In reply to: [327299650](https://github.com/microsoft/vscode-mssql/pull/1301#discussion_r327299650) [](ancestors = 327299650)